inference-engine/src/inference_engine/cnn_network_ngraph_impl.cpp

```diff
@@ -146,7 +146,8 @@ CNNNetworkNGraphImpl::CNNNetworkNGraphImpl(const std::shared_ptr<Function>& nGra
         if (output.second->getPrecision() == Precision::I64) {
             output.second->setPrecision(Precision::I32);
         } else if (output.second->getPrecision() != Precision::FP32 &&
-            output.second->getPrecision() != Precision::I32) {
+            output.second->getPrecision() != Precision::I32 &&
+            output.second->getPrecision() != Precision::U8) {
```

Comment:
Why do we need this additional conditional?

Reply:
Generally my vision is next. This precision fallback policy should be avoided. The tensor precision should match with corresponding precision from nGraph representation. But it's radical change and I'm afraid some scenarios may be broken.
As a small(not such dramatical behavior change) improvement is to keep only U8 output precision as is. 

How does it relate with Loop enabling? The Loop body has a strongly specified output port called "continue_condition". It has BOOL data type. CPU plugin use a BOOL->U8 precision fallback. We expect that this port will keep U8 precision for CNNNetwork representation as well. But it changed to fp32 forcibly... I would like to avoid it.

Reply:
Experiment was failed. GPU plugin doesn't support this case. No time to fix it on gpu plugin side, so I made decision to revert this change.